### PR TITLE
feat(format): .wperf file reader with TLV parsing and crash recovery

### PR DESCRIPTION
**Pull request by adrian-mason**
## Summary
- Add `src/format/reader.rs`: streaming reader for .wperf binary files
- TLV record iteration with forward-compat (skips unknown record types)
- Crash recovery: reads only up to `data_section_end_offset`, gracefully handles missing footer
- DoS protection: rejects TLV payloads exceeding `MAX_PAYLOAD_SIZE` (16MB)
- Footer metadata extraction: parses section table → metadata section → `EVENT_COUNT`/`DROP_COUNT`
- `rewind()` for re-iteration, `read_all_events()` convenience method
- 18 unit tests: roundtrip, empty trace, crash recovery, bad magic, payload overflow, wrong size, unknown types, large trace, error trait coverage

## Authoritative Inputs
- final-design.md §4 (wPRF v1 binary format layout)
- ADR-010 (TLV record structure, section table, DoS protection via MAX_PAYLOAD_SIZE)
- final-design.md §4.1 (64B header: magic, version, data_section_end_offset, section_table_offset)
- PR #72 / PR #73 (40B naturally-aligned wperf_event, footer metadata with EVENT_COUNT/DROP_COUNT)

## Deviations
None

## Task
W2 #11 — .wperf format parser (reader + event deserialization)

## Dependency Checklist
- [x] No new dependencies added

## Review Checklist
- [ ] **Design conformance**: implementation matches the ADRs/specs listed in Authoritative Inputs
- [ ] **Deviations declared**: any spec drift is explicitly listed above with a doc update reference
- [ ] **Code correctness**: logic, error handling, edge cases
- [ ] **Tests**: new/changed code has adequate test coverage
- [ ] **CI green**: all required checks pass

## Test Plan
- [x] `cargo test --lib -- format::reader` — 18/18 pass
- [x] `cargo clippy --all-targets -- -D warnings` — clean
- [x] `cargo fmt --check` — clean
- [x] Full `cargo test` — all pass
- [x] Mutation testing ≥90% kill rate — pass

### CODE REVIEW

File: src/format/reader.rs [hunk 1]
```diff
@@ -0,0 +1,505 @@
...
+            return Ok(Metadata {
+                event_count: None,
+                drop_count: None,
+            });
+        };
+
+        // Read metadata payload
+        self.inner.seek(SeekFrom::Start(offset))?;
+        #[allow(clippy::cast_possible_truncation)] // metadata sections are small
+        let mut buf = vec![0u8; size as usize];
```
[the remark this answers is not in the document]
Comment (adrian-mason) [reply src/format/reader.rs:249]: @gemini-code-assist Good catch — fixed in `a9ec231`. Added `MAX_PAYLOAD_SIZE` guard before metadata allocation.

File: src/format/reader.rs [hunk 2]
```diff
@@ -0,0 +1,505 @@
...
+    /// TLV record has an unexpected payload length for its type.
+    UnexpectedPayloadSize {
+        rec_type: u8,
+        expected: u32,
+        actual: u32,
+    },
+    /// Unknown TLV record type (skipped in lenient mode, error in strict).
+    UnknownRecordType(u8),
+    /// Section table offset points beyond valid data.
+    InvalidSectionTableOffset(u64),
```
[the remark this answers is not in the document]
Comment (adrian-mason) [reply src/format/reader.rs]: @gemini-code-assist Already addressed — `InvalidSectionTableOffset` was removed in an earlier commit.

File: src/format/reader.rs [hunk 3]
```diff
@@ -0,0 +1,505 @@
...
+    /// Unknown record types are skipped (forward-compatible).
+    pub fn next_event(&mut self) -> Result<Option<WperfEvent>, ReaderError> {
+        loop {
+            let pos = self.inner.stream_position()?;
+            if pos >= self.data_end {
+                return Ok(None);
+            }
+
+            // Check if there's enough room for at least a TLV header (5 bytes)
+            if pos + 5 > self.data_end {
```
[the remark this answers is not in the document]
Comment (adrian-mason) [reply src/format/reader.rs]: @gemini-code-assist Fixed in `a9ec231` — now uses `TLV_HEADER_SIZE` constant from writer.rs (made pub).

File: src/format/reader.rs [hunk 4]
```diff
@@ -0,0 +1,505 @@
...
+            // Check the payload fits within the data section
+            let payload_end = self.inner.stream_position()? + u64::from(length);
+            if payload_end > self.data_end {
+                // Truncated record (crash mid-write) — stop here
+                return Ok(None);
+            }
+
+            // Record type 1 = sched event
+            #[allow(clippy::cast_possible_truncation)] // EVENT_SIZE is 40, always fits u32
+            if rec_type == 1 {
```
[the remark this answers is not in the document]
Comment (adrian-mason) [reply src/format/reader.rs]: @gemini-code-assist Fixed in `a9ec231` — now uses `REC_TYPE_SCHED_EVENT` constant from writer.rs (made pub).

File: src/format/reader.rs [hunk 5]
```diff
@@ -0,0 +1,505 @@
...
+        let mut metadata_size = None;
+
+        loop {
+            match writer::read_section_entry(&mut self.inner) {
+                Ok((section_id, offset, size)) => {
+                    // SECTION_ID_METADATA = 3
+                    if section_id == 3 {
+                        metadata_offset = Some(offset);
+                        metadata_size = Some(size);
+                    }
```
[the remark this answers is not in the document]
Comment (adrian-mason) [reply src/format/reader.rs]: @gemini-code-assist Fixed in `a9ec231` — now uses `SECTION_ID_METADATA` constant and breaks after finding the metadata section.
